libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2107,24 +2079,36 @@ smtutil::Expression SMTEncoder::compoundAssignment(Assignment const& _assignment
 	return values.first;
 }
 
-void SMTEncoder::tryCatchAssignment(vector<shared_ptr<VariableDeclaration>> const& _variables, smt::SymbolicVariable const& _rhs)
+void SMTEncoder::expressionToTupleAssignment(vector<shared_ptr<VariableDeclaration>> const& _variables, Expression const& _rhs)
 {
+	auto symbolicVar = m_context.expression(_rhs);
 	if (_variables.size() > 1)
 	{
-		auto const* symbTuple = dynamic_cast<smt::SymbolicTupleVariable const*>(&_rhs);
+		auto symbTuple = dynamic_pointer_cast<smt::SymbolicTupleVariable>(symbolicVar);
 		solAssert(symbTuple, "");
 		auto const& symbComponents = symbTuple->components();
 		solAssert(symbComponents.size() == _variables.size(), "");
+		auto fromType = _rhs.annotation().type;
+		auto tupleType = dynamic_cast<TupleType const*>(fromType);
```

Comment:
Is this always true?

Reply:
I think so. This construction was used in `SMTEncoder::visit(VariableDeclarationStatement const&)` previously.
I think it should be true also for the try/catch case.